pkg/apis/serving/k8s_validation.go

```diff
@@ -267,7 +276,51 @@ func ValidatePodSpec(ps corev1.PodSpec) *apis.FieldError {
 	return errs
 }
 
+//ValidateMultiContainerPorts validates port when specified multiple containers
+func ValidateMultiContainerPorts(containers []corev1.Container) *apis.FieldError {
+	cPort := []int32{}
+	var errs *apis.FieldError
+	for i := range containers {
+		for j := range containers[i].Ports {
+			cPort = append(cPort, containers[i].Ports[j].ContainerPort)
+		}
+	}
+	if len(cPort) == 0 {
+		errs = errs.Also(apis.ErrMissingField("containers.ports.containerPort"))
+	}
+	if len(cPort) > 1 {
+		errs = errs.Also(apis.ErrMultipleOneOf("containers.ports.containerPort"))
+	}
+	return errs
+}
+
+//ValidateMultiContainer validate fields for non serving containers
+func ValidateMultiContainer(containers *corev1.Container, volumes sets.String) *apis.FieldError {
```

Comment:
```suggestion
func ValidateSidecarContainer(containers *corev1.Container, volumes sets.String) *apis.FieldError {
```

I would recommend renaming as `ValidateMultiContainer` makes me believe this is used for _all_ containers within the set and not just the non-serving containers.